.github/workflows/devnet.yaml

```diff
@@ -0,0 +1,85 @@
+# Test the functionality of the spore-sdk packages.
+
+name: Devnet Test
+
+on:
+  pull_request:
+    paths:
+      - contracts/**
+      - lib/**
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+
+jobs:
+  devnet:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout spore-devenv
+        uses: actions/checkout@v4
+        with:
+          repository: sporeprotocol/spore-devenv
+          path: spore-devenv
+
+      - name: Checkout spore-sdk
+        uses: actions/checkout@v4
+        with:
+          repository: sporeprotocol/spore-sdk
+          path: spore-sdk
+
+      - name: Install Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: 20.x
+
+      - uses: pnpm/action-setup@v2
+        name: Install -g pnpm
+        with:
+          version: 8
+          run_install: false
+
+      - name: Get pnpm store directory
+        shell: bash
+        run: |
+          echo "STORE_PATH=$(pnpm store path --silent)" >> $GITHUB_ENV
+
+      - uses: actions/cache@v3
+        name: Setup pnpm cache
+        with:
+          path: ${{ env.STORE_PATH }}
+          key: ${{ runner.os }}-pnpm-store-${{ hashFiles('**/pnpm-lock.yaml') }}
+          restore-keys: |
+            ${{ runner.os }}-pnpm-store-
+
+      - name: Prepare spore-devenv (contracts and stuff)
+        working-directory: spore-devenv
+        run: bash prepare.sh -b $GITHUB_HEAD_REF
```

Comment:
```suggestion
        run: bash prepare.sh -b $GITHUB_REF
```

GITHUB_HEAD_REF is the head ref or source branch of the pull request in a workflow run. This property is **only** set when the event that triggers a workflow run is either pull_request or pull_request_target.

## Suggestions
- Add a `push` event for this workflow, since it only takes 10 minutes.
  Running devnet test for every pushed commit is acceptable.

- use `GITHUB_REF` instead of `GITHUB_HEAD_REF`
  > The fully-formed ref of the branch or tag that triggered the workflow run. For workflows triggered by push, this is the branch or tag ref that was pushed. For workflows triggered by pull_request, this is the pull request merge branch. For workflows triggered by release, this is the release tag created.
  See https://docs.github.com/en/actions/learn-github-actions/variables

Reply:
we used $GITHUB_REF but did not get to our expectation, and for now, $GITHUB_HEAD_REF works fine for us, so it can stay as it is now

Reply:
What do you mean `it did not get to our expectation`?
Is there a GitHub action that contains the error?